rx_signal_processing/dsp.hpp

```diff
@@ -40,25 +40,58 @@ std::vector<cudaDeviceProp> get_gpu_properties();
 void print_gpu_properties(std::vector<cudaDeviceProp> gpu_properties);
 
 
+typedef struct rx_slice
+{
+  double rx_freq;
+  uint32_t slice_id;
+  uint32_t num_ranges;
+  uint32_t beam_count;
+  float first_range; // km
+  float range_sep; // km
+  uint32_t tau_spacing; // us
+
+  struct lag
+  {
+    uint32_t pulse_1;
+    uint32_t pulse_2;
+    uint32_t lag_num;
+    lag(uint32_t pulse_1, uint32_t pulse_2, uint32_t lag_num):
```

Comment:
I don't think I've seen this type of struct before (with the lag(...): pulse_1(...)...{}, what is it?

Reply:
this is a default constructor, not much difference between struct and class in c++ so structs can have default constructors -- Keith